src/highlighter.js

```diff
@@ -148,6 +157,7 @@ export default function (options) {
             isWrapping = false;
             reindexGroups(getContainer());
             mergeAdjacentWrappingNodes(getContainer());
+            unWrapEmptyHighlights();
```

Comment:
1. would that make sense to do this before all the other post-processing steps? (`reindexGroups` and `mergeAdjacentWrappingNodes`)?
2. for consistency shouldn't we pass the container in the same fashion (`getContainer()`) rather than referencing directly `containerSelector` in `getHighlightedNodes`?

Reply:
1. This process should be run right after `mergeAdjacentWrappingNodes`. 
It is possible to get a case when group IDs won't be in numerical sequence(is it a reason why did you ask?), but to be honest I still don’t understand the role of group IDs and I will be really appreciated if you can help me to figure it out.
2. At the beginning, I've started implementation from passing container in the way as we did for `reindexGroups`  and `mergeAdjacentWrappingNodes `, but then I changed it for multiple reasons.
The most important of that, the function will not run recursive(as other processes) that's why there is no sense to pass any parameters and event if the element is going to be pass as parameter, it's required to be wrap in jQuery once again, so that the call will be similar to:
```javascript
$($(containerSelector).get(0)).find(
```
But maybe it makes sense for caching the `getContainer()` and pass it to the post-proccesors functions 🤔

Reply:
I think the purpose of the group Id was to identify highlights nodes that are visually part of the same highlight. 
I don't have a specific error case in mind, I was just wondering if this could have an impact.